### PR TITLE
Update vendored libyaml to 0.2.2

### PR DESCRIPTION
**Pull request by MSP-Greg**
Vendored code had several typos & whitespace 'issues'.  I believe I updated all to match previous code.

### CODE REVIEW

**Comment by hsbt** [comment]
zlib?

**Comment by MSP-Greg** [comment]
Sorry.  Was bouncing between the two.  Updated the title...

**Comment by hsbt** [comment]
@MSP-Greg Can you split commits for libyaml-0.2.2 and CI configuration?

File: ext/psych/yaml/yaml_private.h [hunk 1]
```diff
@@ -1,9 +1,5 @@
-#ifdef RUBY_EXTCONF_H
-#include RUBY_EXTCONF_H
-#endif
```
Comment (mame) [inline ext/psych/yaml/yaml_private.h]: These three lines are needed in my machine (Ubuntu 19.04).

**Comment by MSP-Greg** [comment]
@hsbt

> @MSP-Greg Can you split commits for libyaml-0.2.2 and CI configuration?

Um, you mean split into two PR's?  Or another commit for the psych.gemspec change?

Also, @mame's issue and a merge issue...  Need to fix a little later...

**Comment by hsbt** [comment]
> Um, you mean split into two PR's? Or another commit for the psych.gemspec change?

two PR's.

**Comment by MSP-Greg** [comment]
@hsbt

Sorry for the long delay.  I figured might as well do WIndows on Actions.

See https://github.com/ruby/psych/pull/421

**Comment by MSP-Greg** [comment]
I ran this PR in my fork with PR https://github.com/ruby/psych/pull/424 added, and it passed at:

https://github.com/MSP-Greg/psych/commit/ea222dbf1ef269cc820ff55a1600cd7db313dcec

Sorry for the delay.  I assume this should be included in 2.7.0 release...

**Comment by MSP-Greg** [comment]
Rebased, fixed one typo

**Comment by MSP-Greg** [comment]
Rebased on master to use updated CI.

* **Ubuntu**
  Ruby 2.6 and later compile without warnings
  JRuby - warning on both jruby & jruby-head

* **macOS**
  Ruby compiles without warnings

* **Windows**
  All builds compile without warnings

**Comment by MSP-Greg** [comment]
Closing in favor of PR https://github.com/ruby/psych/pull/436

**Pull request closed without merging**